eventsourcing/src/main/java/org/axonframework/eventsourcing/annotation/AnnotationBasedEventSourcingEntityRepository.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.annotation;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.ConstructorUtils;
+import org.axonframework.common.annotation.AnnotationUtils;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.eventsourcing.AnnotationBasedEventStateApplier;
+import org.axonframework.eventsourcing.AsyncEventSourcingRepository;
+import org.axonframework.eventsourcing.CriteriaResolver;
+import org.axonframework.eventsourcing.eventstore.AsyncEventStore;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+import org.axonframework.modelling.repository.AsyncRepository;
+import org.axonframework.modelling.repository.ManagedEntity;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static org.axonframework.common.ConstructorUtils.factoryForTypeWithOptionalArgumentInstance;
+
+/**
+ * Implementation of a {@link AsyncRepository} that configures an entity based on the configuration provided by the
+ * {@link EventSourcedEntity} annotation. Please refer to the {@link EventSourcedEntity} annotation for more information
+ * on the configuration options.
+ *
+ * @param <ID> The type of the identifier of the entity.
+ * @param <T>  The type of the entity.
+ */
+public class AnnotationBasedEventSourcingEntityRepository<ID, T> implements AsyncRepository.LifecycleManagement<ID, T> {
```

Comment:
I am wondering. Is this a concern of the `Repository`, or should this be a different type of  `EntityBuilder`?
Granted, I only introduced that last Friday...

Although we worked in parallel and didn't discuss this together in enough detail, I figured it worthwhile to touch on the subject regardless. This `Repository` implementation only really does some constructor logic. Besides that, it's just a means to construct, and delegate, an `AsyncEventSourcingRepository`.